Sources/Core/Transaction/EthereumBloomFilter.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  EthereumBloomFilter.swift
+//  web3swift
+//
+//  Created by Alex Vlasov.
+//  Copyright © 2018 Alex Vlasov. All rights reserved.
+//
+
+import Foundation
+import BigInt
+import CryptoSwift
+
+/// A wrapper around a set of bytes that represent a Bloom filter.
+/// Similar implementation in Go language can be found [here in bloom9.go](https://github.com/ethereum/go-ethereum/blob/master/core/types/bloom9.go).
+///
+/// Bloom filter is used to reduce the cost in terms of memory consumption during the process of searching
+/// Bloom filter can be calculated for any set of data. In case of Ethereum blockchain it could be a set of addresses, event topics  etc.
+///
+/// A definition of [Bloom filter](https://en.wikipedia.org/wiki/Bloom_filter#:~:text=A%20Bloom%20filter%20is%20a,a%20member%20of%20a%20set.).
+public struct EthereumBloomFilter {
+    static let mask = BigUInt(2047)
+    /// Bloom filter.
+    public var bytes = Data(repeatElement(UInt8(0), count: 256))
+    public init?(_ biguint: BigUInt) {
+        guard let data = biguint.serialize().setLengthLeft(256) else {return nil}
+        bytes = data
+    }
+    public init() {}
+    public init(_ data: Data) {
+        let padding = Data(repeatElement(UInt8(0), count: 256 - data.count))
+        bytes = padding + data
+    }
+    public func asBigUInt() -> BigUInt {
+        BigUInt(bytes)
+    }
+}
+
+extension EthereumBloomFilter {
+
+    // MARK: - Bloom filter calculation functions
+    /// Calculates Bloom filter from Keccak-256 calculated from given `number`.
+    /// - Parameter number: some number to calculate filter from.
+    /// - Returns: Bloom filter.
+    static func bloom9(_ number: BigUInt) -> BigUInt {
+        bloom9(number.serialize())
+    }
+
+    /// Calculates Bloom filter from Keccak-256 calculated from given `data`.
+    /// - Parameter data: some data to calculate filter from, e.g. event's topic or address of a smart contract.
+    /// - Returns: Bloom filter.
+    static func bloom9(_ data: Data) -> BigUInt {
+        // TODO: update to match this implementation https://manbytesgnu.com/eth-log-bloom.html
+        // TODO: it will increase performance.
+        let b = data.sha3(.keccak256)
```

Comment:
A few comments left here with a link to the implementation of a bloom filter that should have better performance.
I'll open an issue so that we keep a record of that.